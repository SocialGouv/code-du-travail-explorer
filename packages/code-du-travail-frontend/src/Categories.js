import React from "react";
import PropTypes from "prop-types";
import { Link } from "../routes";

import { Container, Categories as CategoriesWrapper } from "@cdt/ui";

import themes from "@cdt/data/dataset/themes-front.json";
export default class Categories extends React.Component {
  static propTypes = {
    themes: PropTypes.arrayOf(
      PropTypes.shape({
        title: PropTypes.string.isRequired,
        slug: PropTypes.arrayOf(PropTypes.string).isRequired,
        icon: PropTypes.string.isRequired,
        text: PropTypes.string.isRequired
      })
    ),
    isRoot: PropTypes.bool,
    title: PropTypes.string
  };

  static defaultProps = {
    themes,
    isRoot: true,
    title: "Retrouvez nos réponses thématiques"
  };

  render() {
    const { title, themes, isRoot } = this.props;
    return (
      (themes.length && (
        <Container>
          {title && (
            <h2
              style={{ marginTop: 20, marginBottom: 40, textAlign: "center" }}
            >
              {title}
            </h2>
          )}
          <CategoriesWrapper>
            {themes.map(theme => (
              <Category
                key={theme.slug + theme.title}
                small={!isRoot}
                {...theme}
              />
            ))}
          </CategoriesWrapper>
          <br />
          <br />
        </Container>
      )) ||
      null
    );
  }
}

const Category = ({ title, text, slug, small, icon }) => (
  <li
    className={`categories__list-item ${(small &&
      "categories__list-item--small") ||
      ""}`}
  >
    <Link route="theme" params={{ slug: slug || "/" }}>
      <a title={title}>
        <figure>
          <img src={icon} alt={title} />
        </figure>
        <div className="categories__list-item-content">
          <h3>{title}</h3>
          <p>{text}</p>
        </div>
      </a>
    </Link>
  </li>
);

Category.propTypes = {
  title: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  slug: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.arrayOf(PropTypes.string)
  ]).isRequired,
  small: PropTypes.bool,
  icon: PropTypes.string
};
Category.defaultProps = {
  small: false,
  icon: "/static/assets/icons/chat.svg"
};
