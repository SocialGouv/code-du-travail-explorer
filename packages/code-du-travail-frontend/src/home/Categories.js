import React from "react";
import PropTypes from "prop-types";
import { Link } from "../../routes";

import {
  Container,
  Categories as CategoriesWrapper,
  Category as CategoryItem
} from "@cdt/ui";

import themes from "@cdt/data/dataset/themes-front.json";

export default class Categories extends React.Component {
  static propTypes = {
    themes: PropTypes.arrayOf(
      PropTypes.shape({
        title: PropTypes.string.isRequired,
        slug: PropTypes.arrayOf(PropTypes.string).isRequired,
        icon: PropTypes.string.isRequired,
        text: PropTypes.string.isRequired
      })
    ),
    isRoot: PropTypes.bool,
    title: PropTypes.string
  };

  static defaultProps = {
    themes,
    isRoot: true,
    title: "Retrouvez nos réponses thématiques"
  };

  render() {
    const { title, themes, isRoot } = this.props;
    return (
      (themes.length && (
        <section>
          <Container>
            {title && (
              <h2
                style={{ marginTop: 20, marginBottom: 40, textAlign: "center" }}
              >
                {title}
              </h2>
            )}
            <CategoriesWrapper>
              {themes.map(({ slug, title, text, icon }) => (
                <CategoryItem key={slug + title} small={!isRoot}>
                  <Link route="theme" params={{ slug: slug || "/" }}>
                    <a title={title}>
                      <Category title={title} text={text} icon={icon} />
                    </a>
                  </Link>
                </CategoryItem>
              ))}
            </CategoriesWrapper>
          </Container>
        </section>
      )) ||
      null
    );
  }
}

const Category = ({ title, text, icon }) => (
  <React.Fragment>
    <figure>
      <img src={icon} alt={title} />
    </figure>
    <div>
      <h3>{title}</h3>
      <p>{text}</p>
    </div>
  </React.Fragment>
);

Category.propTypes = {
  title: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  icon: PropTypes.string
};
Category.defaultProps = {
  icon: "/static/assets/icons/chat.svg"
};
