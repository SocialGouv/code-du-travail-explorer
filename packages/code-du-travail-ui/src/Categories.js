import React from "react";
import PropTypes from "prop-types";

const Categories = ({ className = "", ...props }) => (
  <div className="categories">
    <ul className={`categories__list ${className}`} {...props} />
  </div>
);

Categories.propTypes = {
  className: PropTypes.string
};

export default Categories;
