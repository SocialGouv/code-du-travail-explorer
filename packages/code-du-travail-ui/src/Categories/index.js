import styled from "styled-components";
import { spacing } from "../theme";

const Categories = styled.ul`
  display: flex; /* Flex layout so items have equal height. */
  flex-wrap: wrap;
  justify-content: flex-start;
  align-content: flex-start;
  list-style-type: none;
  padding: 0;
  /* Use negative margins on sides to create gutters that do not also
    create a gutter at the edges of the container. */
  margin: ${-1 * spacing.small};
`;

export default Categories;
