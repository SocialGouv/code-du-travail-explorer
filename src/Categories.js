import React from "react";
import { Link } from "../routes";

import { Container, Categories } from "@socialgouv/code-du-travail-ui";

import themes from "./data/themes2";

const Category = ({
  title,
  text,
  slug,
  small = false,
  icon = "/static/assets/icons/chat.svg"
}) => (
  <li
    className={`categories__list-item ${(small &&
      "categories__list-item--small") ||
      ""}`}
  >
    <Link route="theme" params={{ slug: slug || "/" }}>
      <a title={title}>
        <figure>
          <img src={icon} alt={title} />
        </figure>
        <div className="categories__list-item-content">
          <h3>{title}</h3>
          <p>{text}</p>
        </div>
      </a>
    </Link>
  </li>
);

class _Categories extends React.Component {
  render() {
    const { title, themes, isRoot } = this.props;
    return (
      (themes.length && (
        <Container>
          {title && (
            <h2
              style={{ marginTop: 20, marginBottom: 40, textAlign: "center" }}
            >
              {title}
            </h2>
          )}
          <Categories>
            {themes.map(theme => (
              <Category
                key={theme.slug + theme.title}
                small={!isRoot}
                {...theme}
              />
            ))}
          </Categories>
          <br />
          <br />
        </Container>
      )) ||
      null
    );
  }
}

_Categories.defaultProps = {
  themes,
  title: "Retrouvez nos réponses thématiques"
};

export default _Categories;
