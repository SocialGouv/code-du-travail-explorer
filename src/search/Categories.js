import React from "react";

class Categories extends React.Component {
  handleClick = (event, query) => {
    event.preventDefault();
    this.props.urlUpdate(encodeURI(query));
  };

  render() {
    return (
      <section className="section-light">
        <div className="container">
          <div className="categories">
            <header className="center">
              <h2 className="no-margin">Retrouvez nos réponses thématiques</h2>
            </header>
            <ul className="categories__list">
              <li className="categories__list-item">
                <a
                  href="#"
                  onClick={e => this.handleClick(e, "obligations et embauche")}
                >
                  <img src="/static/assets/icons/handshake.svg" alt="" />
                  <h3>contrat de travail et embauche</h3>
                </a>
              </li>
              <li className="categories__list-item">
                <a
                  href="#"
                  onClick={e =>
                    this.handleClick(e, "durée de travail et congés")
                  }
                >
                  <img src="/static/assets/icons/time.svg" alt="" />
                  <h3>Durée de travail et congés</h3>
                </a>
              </li>
              <li className="categories__list-item">
                <a href="#" onClick={e => this.handleClick(e, "rémunération")}>
                  <img src="/static/assets/icons/remuneration.svg" alt="" />
                  <h3>Rémunération</h3>
                </a>
              </li>
              <li className="categories__list-item">
                <a
                  href="#"
                  onClick={e => this.handleClick(e, "santé et sécurité")}
                >
                  <img src="/static/assets/icons/shield.svg" alt="" />
                  <h3>Santé et sécurité</h3>
                </a>
              </li>
              <li className="categories__list-item">
                <a
                  href="#"
                  onClick={e => this.handleClick(e, "rupture du contrat de travail")}
                >
                  <img src="/static/assets/icons/file-3.svg" alt="" />
                  <h3>Rupture du contrat de travail </h3>
                </a>
              </li>
              <li className="categories__list-item">
                <a href="#" onClick={e => this.handleClick(e, "formation")}>
                  <img src="/static/assets/icons/hiring-1.svg" alt="" />
                  <h3>Formation</h3>
                </a>
              </li>
              <li className="categories__list-item">
                <a
                  href="#"
                  onClick={e => this.handleClick(e, "dialogue social")}
                >
                  <img src="/static/assets/icons/chat.svg" alt="" />
                  <h3>Dialogue social</h3>
                </a>
              </li>
              <li className="categories__list-item">
                <a
                  href="#"
                  onClick={e => this.handleClick(e, "maladie et inaptitude")}
                >
                  <img src="/static/assets/icons/certificate.svg" alt="" />
                  <h3>Maladie et inaptitude</h3>
                </a>
              </li>
            </ul>
          </div>
        </div>
      </section>
    );
  }
}

export default Categories;
