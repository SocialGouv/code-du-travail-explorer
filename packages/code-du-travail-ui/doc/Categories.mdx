---

---

import { Playground, PropsTable } from "docz";
import { Categories, Category  } from "../src";


## \<Category/\>

<Playground>
  <Categories>
    <Category icon="https://rawgit.com/SocialGouv/code-du-travail-html/master/docs/assets/icons/handshake.svg" title="Embauche et contrat"/>
    <Category icon="https://rawgit.com/SocialGouv/code-du-travail-html/master/docs/assets/icons/time.svg" title="Durée de travail et congés"/>
    <Category icon="https://rawgit.com/SocialGouv/code-du-travail-html/master/docs/assets/icons/coins.svg" title="Rémunération"/>
    <Category icon="https://rawgit.com/SocialGouv/code-du-travail-html/master/docs/assets/icons/shield.svg" title="Santé et sécurité"/>
  </Categories>
</Playground>

<PropsTable of={Category} />
