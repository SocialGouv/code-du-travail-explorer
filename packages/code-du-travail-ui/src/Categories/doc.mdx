---
name: Categories
menu: Component
---
import "@cdt/css";
import { Playground, Props } from "docz";
import Cell, {Category} from "../Category";
import Categories from ".";

# Categories

<Playground>
  <Categories>
    <Cell>
      <Category
        icon="https://rawgit.com/SocialGouv/code-du-travail-html/master/docs/assets/icons/handshake.svg"
        title="Embauche et contrat"
        text="c'est trop bien le texte"
      />
    </Cell>
    <Cell>
      <Category
        icon="https://rawgit.com/SocialGouv/code-du-travail-html/master/docs/assets/icons/time.svg"
        title="Durée de travail et congés"
      />
    </Cell>
    <Cell>
      <Category
        icon="https://rawgit.com/SocialGouv/code-du-travail-html/master/docs/assets/icons/coins.svg"
        title="Rémunération"
      />
    </Cell>
    <Cell>
      <Category
        icon="https://rawgit.com/SocialGouv/code-du-travail-html/master/docs/assets/icons/shield.svg"
        title="Santé et sécurité"
      />
    </Cell>
  </Categories>
  # Small categories
  <Categories>
    <Cell>
      <Category
        small
        icon="https://rawgit.com/SocialGouv/code-du-travail-html/master/docs/assets/icons/handshake.svg"
        title="Embauche et contrat"
       />
    </Cell>
    <Cell>
      <Category
        small
        icon="https://rawgit.com/SocialGouv/code-du-travail-html/master/docs/assets/icons/time.svg"
        title="Durée de travail et congés"
      />
    </Cell>
    <Cell>
      <Category
        small
        icon="https://rawgit.com/SocialGouv/code-du-travail-html/master/docs/assets/icons/coins.svg"
        title="Rémunération"
      />
    </Cell>
    <Cell>
      <Category
        small
        icon="https://rawgit.com/SocialGouv/code-du-travail-html/master/docs/assets/icons/shield.svg"
        title="Santé et sécurité"
      />
    </Cell>
  </Categories>
</Playground>

<Props of={Category} />
