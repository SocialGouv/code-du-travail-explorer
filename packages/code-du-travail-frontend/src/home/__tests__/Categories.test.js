import React from "react";
import { render } from "react-testing-library";
import Categories from "../Categories";

describe("<Categories />", () => {
  it("should render", () => {
    const { container } = render(<Categories />);
    expect(container).toMatchSnapshot();
  });
});
