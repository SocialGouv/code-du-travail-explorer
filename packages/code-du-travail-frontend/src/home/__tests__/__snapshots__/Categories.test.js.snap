// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Categories /> should render 1`] = `
<div>
  <section>
    <div
      class="container"
    >
      <h2
        style="margin-top: 20px; margin-bottom: 40px; text-align: center;"
      >
        Retrouvez nos réponses thématiques
      </h2>
      <div
        class="categories"
      >
        <ul
          class="categories__list "
        >
          <li
            class="categories__list-item"
          >
            <a
              href="/themes/rupture-de-contrat"
              title="Rupture de contrat"
            >
              <figure>
                <img
                  alt="Rupture de contrat"
                  src="/static/assets/icons/file-3.svg"
                />
              </figure>
              <div>
                <h3>
                  Rupture de contrat
                </h3>
                <p>
                  Licenciement, démission, rupture conventionnelle…
                </p>
              </div>
            </a>
          </li>
          <li
            class="categories__list-item"
          >
            <a
              href="/themes/embauche-et-contrat"
              title="Contrat de travail et embauche"
            >
              <figure>
                <img
                  alt="Contrat de travail et embauche"
                  src="/static/assets/icons/handshake.svg"
                />
              </figure>
              <div>
                <h3>
                  Contrat de travail et embauche
                </h3>
                <p>
                  Obligations liées à l'embauche et vie du contrat de travail
                </p>
              </div>
            </a>
          </li>
          <li
            class="categories__list-item"
          >
            <a
              href="/themes/duree-du-travail-et-conges"
              title="Durée de travail et congés"
            >
              <figure>
                <img
                  alt="Durée de travail et congés"
                  src="/static/assets/icons/time.svg"
                />
              </figure>
              <div>
                <h3>
                  Durée de travail et congés
                </h3>
                <p>
                  heures supplémentaires, congés payés, durée maximale du travail...
                </p>
              </div>
            </a>
          </li>
          <li
            class="categories__list-item"
          >
            <a
              href="/themes/remuneration"
              title="Rémunération"
            >
              <figure>
                <img
                  alt="Rémunération"
                  src="/static/assets/icons/remuneration.svg"
                />
              </figure>
              <div>
                <h3>
                  Rémunération
                </h3>
                <p>
                  Salaire, primes, avantages, bulletin de salaire...
                </p>
              </div>
            </a>
          </li>
          <li
            class="categories__list-item"
          >
            <a
              href="/themes/sante-et-securite"
              title="Santé et sécurité"
            >
              <figure>
                <img
                  alt="Santé et sécurité"
                  src="/static/assets/icons/shield.svg"
                />
              </figure>
              <div>
                <h3>
                  Santé et sécurité
                </h3>
                <p>
                  conditions de travail, surveillance médicale...
                </p>
              </div>
            </a>
          </li>
          <li
            class="categories__list-item"
          >
            <a
              href="/themes/formation"
              title="Formation"
            >
              <figure>
                <img
                  alt="Formation"
                  src="/static/assets/icons/hiring-1.svg"
                />
              </figure>
              <div>
                <h3>
                  Formation
                </h3>
                <p>
                  Evolution de carrière, VAE
                </p>
              </div>
            </a>
          </li>
          <li
            class="categories__list-item"
          >
            <a
              href="/themes/contentieux-et-coercitif"
              title="prud’hommes, inspection du travail, procédures..."
            >
              <figure>
                <img
                  alt="prud’hommes, inspection du travail, procédures..."
                  src="/static/assets/icons/chat.svg"
                />
              </figure>
              <div>
                <h3>
                  prud’hommes, inspection du travail, procédures...
                </h3>
                <p>
                  Les syndicats à votre service
                </p>
              </div>
            </a>
          </li>
          <li
            class="categories__list-item"
          >
            <a
              href="/themes/maladies-et-inaptitudes"
              title="Maladie et inaptitude"
            >
              <figure>
                <img
                  alt="Maladie et inaptitude"
                  src="/static/assets/icons/certificate.svg"
                />
              </figure>
              <div>
                <h3>
                  Maladie et inaptitude
                </h3>
                <p>
                  inaptitude,arrets de travail, accident du travail /maladie professionnelle...
                </p>
              </div>
            </a>
          </li>
        </ul>
      </div>
    </div>
  </section>
</div>
`;
